tests/net/ethernet_mgmt/src/main.c

```diff
@@ -105,6 +105,8 @@ static int eth_fake_set_config(struct device *dev,
 				     sizeof(ctx->mac_address),
 				     NET_LINK_ETHERNET);
 		break;
+	default:
```

Comment:
Is the testing for ETHERNET_CONFIG_TYPE_FILTER missing here?

Reply:
Yes, since the filtering is not going to be exposed through ethernet net mgmt, I don't see why that test should test for device driver api filter stuff.

Reply:
I probably missed the point here, but if we are introducing new config option we should test it like we are testing the other config options.

Reply:
This test "ethernet_mgmt" is testing the net mgmt interface for ethernet, not the ethernet config stuff. (it just happened that, when ethernet mgmt was introduced, it was a 1:1 with ethernet config, but that's already diverging.)

The new ethernet config option I added is not meant to be accessed via net mgmt, no need to test it.

Plus, if the point was actually to test ethernet config options, there would be no point "testing" these by emulating the feature since in real usage, each driver will implement it (if supported) depending on their datasheet.

That test was only meant to test ethernet mgmt interface, nothing more.

Reply:
Well, I think this config thing needs to be unit tested somehow. It looks quite natural to put such tests into tests/net/ethernet_mgmt/src/main.c or we can certainly create a new file for these if needed. The tests are basically very dummy tests as we are not going to use the config options, the tests just need to verify that the values are passed properly.

Reply:
No, this test tests subsys/net/ip/l2/ethernet/ethernet_mgmt.c not the ethernet device driver API. Because that's generic core L2 code. Ethernet device driver API implementation will be specific to every ethernet device driver: it makes no sense having a test for it.

Reply:
I am baffled now. Don't we need to have similar tests for passing the filtering data as what were done in commit b71d6babfbf